### PR TITLE
Drv 599 complete 3rd party auth support

### PR DESCRIPTION
**Pull request by vadimLF**
The following functions were added:
- CurrentIdentity
- CurrentToken
- HasCurrentIdentity
- HasCurrentToken

### CODE REVIEW

**Review by parkhomenko — DISMISSED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,11 @@
-# v3.0.0 (August, 2020) [current]
+# v4.0.0 (April, 2021) [current]
+
+- Add streaming feature
+- 3rd party Auth support
```
Comment (faunaee) [inline CHANGELOG.md:4]:
```suggestion
- Add third-party authentication functions: AccessProvider, AccessProviders, CreateAccessProvider,
  CurrentIdentity, CurrentToken, HasCurrentIdentity, HasCurrentToken.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,11 @@
-# v3.0.0 (August, 2020) [current]
+# v4.0.0 (April, 2021) [current]
+
+- Add streaming feature
```
Comment (faunaee) [inline CHANGELOG.md:3]:
```suggestion
- Add document streaming.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,11 @@
...
+- 3rd party Auth support
+- Omitempty tag support
```
Comment (faunaee) [inline CHANGELOG.md:5]:
```suggestion
- Add "omitempty" support for JSON encodings.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,11 @@
...
+- Runtime partners info via header support
```
Comment (faunaee) [inline CHANGELOG.md:6]:
```suggestion
- Add support for partners info via request headers.
```

File: README.md [hunk 5]
```diff
@@ -1,18 +1,17 @@
...
 A Go lang driver for [FaunaDB](https://fauna.com/).
 
 ## Supported Go Versions
 
 Currently, the driver is tested on:
-- 1.11
-- 1.12
 - 1.13
 - 1.14
+- 1.15
```
Comment (faunaee) [inline README.md:14]: Do we want to include 1.16 here? I've been running my doc example tests for the new third-party auth functions on 1.16.

**Review by faunaee — CHANGES_REQUESTED** [review]
I've made a few suggestions for edits to the README file.

Otherwise, looks good. The new functions included in this PR and the ones previously added for the V4 driver appear to work well.

File: CHANGELOG.md (hunk 1 above)
Comment (vadimLF) [reply CHANGELOG.md:4]: Done.

File: CHANGELOG.md (hunk 2 above)
Comment (vadimLF) [reply CHANGELOG.md:3]: Done.

File: README.md (hunk 5 above)
Comment (vadimLF) [reply README.md:14]: Done.

File: README.md [hunk 6]
```diff
@@ -72,11 +72,52 @@ func main() {
...
+		case f.ErrorEventT:
+			// do smth on error event
+			subscription.Close() // if you want to close streaming on errors
+		}
+	}
+}
+```
+
 For more information about FaunaDB query language, consult our query language
 [reference documentation](https://docs.fauna.com/fauna/current/reference/queryapi/).
```
Comment (faunaee) [inline README.md:117]:
```suggestion
For more information about Fauna Query Language (FQL), consult our query language
[reference documentation](https://docs.fauna.com/fauna/current/api/fql/).
```

**Review by faunaee — CHANGES_REQUESTED** [review]
Looks pretty good, overall. I made one suggestion unrelated to your changes.

It would be great to add an example of using `omitempty` in the README file.

File: README.md (hunk 6 above)
Comment (updatus) [reply README.md]: Done.

File: README.md [hunk 7]
```diff
@@ -72,11 +72,92 @@ func main() {
...
+		PaymentPointer *float64    `fauna:"paymentPointer,omitempty"`
+	}
+	_, err := dbClient.Query(
+		f.Create(f.Collection("categories"), f.Obj{"data": OmitStruct{Name: "John", Age: 0}}))
+	if err != nil {
+		panic(err)
+	}
+}
+```
+###Result (empty/zero fields will be ignored):
```
Comment (faunaee) [inline README.md:144]:
```suggestion

**Result (empty/zero fields will be ignored):**

```

As a heading there needs to be a space between `#` and the text. But this result is part of the `omitempty` example, so a new section title is inappropriate.

**Review by faunaee — CHANGES_REQUESTED** [review]
One last change, to fix the markup for the `omitempty` heading.

**Review by faunaee — DISMISSED** [review]
LGTM! 👍

File: .circleci/config.yml [hunk 8]
```diff
@@ -58,6 +58,21 @@ commands:
...
+    steps:
+      - build_and_test
+
+  core-nightly-1-16-3:
+    executor:
+      name: core
+      go_version: "1.16.3"
+      version: nightly
+    steps:
+      - build_and_test
```
Comment (parkhomenko) [inline .circleci/config.yml:75]: nit: missed a new line

**Review by parkhomenko — APPROVED** [review]

**Review by parkhomenko — APPROVED** [review]

**Review by tigger9flow — APPROVED** [review]

**Pull request merged**